### PR TITLE
conversions: [torch_dist --> torch]  &  [core --> HF]

### PR DESCRIPTION
**Pull request by dhia680**
This PR extends `tools/checkpoint` to support the conversion of a Megatron core checkpoint to a LlamaForCausalLM (HF). 
Run `tools/checkpoint/do-convert.sh` with your updated paths/variables to perform both conversions.

### CODE REVIEW

**Comment by TJ-Solergibert** [comment]
Proposed changes:
1. Move `tools/checkpoint/do-convert.sh` to `scripts/conversion`
2. Convert `tools/checkpoint/do-convert.sh` to a Slurm batched script
3. Make the script such as we input a precise checkpoint folder like in the submission scripts (`CKPT_DIR`) and a ITERATION. If no iteration provided, get the last one from `latest_checkpointed_iteration.txt` as we do during training
4. Store the checkpoints in `$CKPT_DIR/hf_checkpoints/$ITERATION` (Same structure as the checkpoints stored during training)
5. Include this in the readme
6. Check the generations of the converted hf checkpoints vs the megatron ones with the endpoint inference feature match. And compare the logits despite they dont fully match.

**Comment by AleHD** [comment]
Closing in favour of #52 (my bad)
